liboptv/tests/check_orientation.c

```diff
@@ -19,21 +19,210 @@
+    int nfix, i, k=0, pnr, ix, iy, iz, pt_id;
+    int eps, correct_eps = 25;
+    double xp, yp;
+    FILE *fpp;
+    
+    pt_id = 0;
+    for (ix = 0; ix < 4; ix++) {
+        for (iy = 0; iy < 4; iy++) {
+             for (iz = 0; iz < 4; iz++) {
+                vec_set(fix[pt_id], (ix * 10)-60, iy * 5, iz*5);
+                pt_id ++;
+                }
+            } 
+        }
+
+    /* read the orientation and the parameters */
+    // char ori_file[] = "testing_fodder/cal/cam1.tif.ori";
+    char ori_file[] = "testing_fodder/cal/sym_cam1.tif.ori";
+    char add_file[] = "testing_fodder/cal/cam1.tif.addpar";
+
+    fail_if((cal = read_calibration(ori_file, add_file, NULL)) == 0);
+    fail_if((cpar = read_control_par("testing_fodder/parameters/ptv.par"))== 0);
+
+
+    /* fake the pix points by back-projection */
+    for (i=0; i<64; i++){
+        img_coord (fix[i], cal, cpar->mm, &xp, &yp);
+        metric_to_pixel (&(pix[i].x), &(pix[i].y), xp, yp, cpar);
+        pix[i].pnr = i;
+        // printf("%d %lf %lf %lf\n", i, fix[i][0],fix[i][1],fix[i][2]);
+        // printf("%d, %lf %lf\n", pix[i].pnr, pix[i].x,pix[i].y);
+        
+    }
+
+    /* read orientation parameters */
+    fail_if((opar = read_orient_par("testing_fodder/parameters/orient.par"))== 0);
+    
+    /* perturb the orientation, orient, compare */
+    cal->ext_par.x0 -= 15.0;
+    cal->ext_par.y0 += 15.0;
+    cal->ext_par.z0 -= 115.0;
+    cal->ext_par.omega -= 0.5;
+    cal->ext_par.phi += 0.5;
+    cal->ext_par.kappa += 0.5;
+    
+    fail_if(orient (cal, cpar, 64, fix, pix, opar, &sigmabeta) == NULL);
+    fail_if((org_cal = read_calibration(ori_file, add_file, NULL)) == NULL);
+    fail_unless (fabs(cal->ext_par.x0 - org_cal->ext_par.x0) +
+            fabs(cal->ext_par.y0 - org_cal->ext_par.y0) +
+            fabs(cal->ext_par.z0 - org_cal->ext_par.z0) +
+            fabs(cal->ext_par.omega - org_cal->ext_par.omega) +
+            fabs(cal->ext_par.phi - org_cal->ext_par.phi) +
+            fabs(cal->ext_par.kappa - org_cal->ext_par.kappa) < 1E-6);
+
+    /* perturb the orientation with internal parameters too*/
+    cal->ext_par.x0 -= 15.0;
+    cal->ext_par.y0 += 15.0;
+    cal->ext_par.z0 -= 115.0;
+    cal->ext_par.omega -= 0.5;
+    cal->ext_par.phi += 0.5;
+    cal->ext_par.kappa += 0.5;
+    cal->int_par.cc -= 5;
+    cal->int_par.xh += 1.0;
+    cal->int_par.yh -= 1.0;
+    
+    opar->ccflag = 1;
+    opar->xhflag = 1;
+        
+    fail_if(orient (cal, cpar, 64, fix, pix, opar, &sigmabeta) == NULL);
+    fail_unless (fabs(fabs(cal->ext_par.x0 - org_cal->ext_par.x0) +
+            fabs(cal->ext_par.y0 - org_cal->ext_par.y0) +
+            fabs(cal->ext_par.z0 - org_cal->ext_par.z0) +
+            fabs(cal->ext_par.omega - org_cal->ext_par.omega)/180 +
+            fabs(cal->ext_par.phi - org_cal->ext_par.phi)/180 +
+            fabs(cal->ext_par.kappa - org_cal->ext_par.kappa)/180 - 19.495073)< 1E-6);
+            
+    /* print results */
```

Comment:
yes please. although if we remove the print from the liboptv (that was the idea), we need an alternative interface for the user to show the results of the convergence. as you know, this is crucial in this kind of interactive work.